### ISSUE TITLE
Error handling and session recovery

### ISSUE THREAD

**Issue by darkowic** [continuedev/continue#3185]
### Validations

- [X] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [X] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

The whole session is lost when any error occurs. Its really bad experience because whatever I already wrote in the chat is gone. This happens many many times and it can not be recovered anyhow — I must start from scratch.

![Screenshot 2024-12-04 at 16 58 24](https://github.com/user-attachments/assets/cd7a309b-3516-495e-99b4-84d4fc8fdbe9)

### Solution

Better error handling and always preserving what user already provided.

**Comment by Patrick-Erichsen** [comment]
Yeah this is a huge pain for me as well. We're doing a lot of bugfixes this month and will prioritize this!

**Comment by RomneyDa** [comment]
Better streaming errors and recovery has been implemented throughout several PRs.
